paddle/fluid/operators/mkldnn/fc_mkldnn_op.cc

```diff
@@ -54,6 +54,19 @@ class FCPrimitiveFactory {
       return;
     }  // Otherwise, create a new one.
 
+    auto in_col_dims = ctx.Attr<int>("in_num_col_dims");
+    if (in_col_dims == 2) {
+      PADDLE_ENFORCE_EQ(
+          input->dims().size(), 3,
+          platform::errors::Unimplemented(
+              "DNNL FC only supports in_num_col_dims equal to 2 when "
+              "3 dim input is provided."));
+    } else if (in_col_dims > 2) {
+      PADDLE_THROW(platform::errors::Unimplemented(
+          "DNNL FC doesn't support in_num_col_dims paramter to be higher than "
+          "2."));
+    }
```

Comment:
What about structuring this part like this:
```
PADDLE_ENFORCE_LE(in_col_dims, 2, ...);
if (in_col_dims == 2) {
  PADDLE_ENFORCE_EQ(input->dims().size(), 3, ...);
}
```
?

Reply:
Sure, that would definitely make it look much clearer.